### PR TITLE
Issue #7988 Typo changed also added comments for each index, getters and setters.

### PR DESCRIPTION
**Pull request by arshadpkm**
### Description
Issue #7988 Typo changed also added comments for each index, getters and setters.

### Fixed Issues (if relevant)
1. magento/magento2#7988: Typo in variable name: billingAddressFromData vs. billingAddressFormData 

### Manual testing scenarios
1. Not applicable

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=9484) <br/>All committers have signed the CLA.

**Comment by adragus-inviqa** [comment]
I'm not sure comments add much value here. Comments usually don't. I'd rather you take them out.

What happens if the logic changes and the values are not taken from local storage? You'd have to update the comments, too. Suddently, you have two things to maintain.

File: app/code/Magento/Checkout/view/frontend/web/js/checkout-data.js [hunk 1]
```diff
@@ -33,20 +33,24 @@ define([
...
         checkoutData = {
-            'selectedShippingAddress': null,
-            'shippingAddressFromData': null,
-            'newCustomerShippingAddress': null,
-            'selectedShippingRate': null,
-            'selectedPaymentMethod': null,
-            'selectedBillingAddress': null,
-            'billingAddressFormData': null,
-            'newCustomerBillingAddress': null
+            'selectedShippingAddress': null, // Selected shipping address pullled from local storage (Persistence)
```
Comment (okorshenko) [inline app/code/Magento/Checkout/view/frontend/web/js/checkout-data.js]: 1. remove space between // and Selected (fix all cases)
2. replace local storage (Persistence) and  local storage --> persistence storage

File: app/code/Magento/Checkout/view/frontend/web/js/checkout-data.js [hunk 2]
```diff
@@ -33,20 +33,24 @@ define([
...
+            'newCustomerBillingAddress': null, // Billing address pullled from local storage for new customer (Persistence)
+            'validatedEmailValue': null, // Validated email address from local storage (Persistence)
+            'inputFieldEmailValue' : null // Email input field value from local storage (Persistence)
         };
         saveData(checkoutData);
     }
 
     return {
         /**
+         * Setting the selected shipping address pulled from local storage
```
Comment (okorshenko) [inline app/code/Magento/Checkout/view/frontend/web/js/checkout-data.js]: add . at the end of the comment. (all cases)

**Review by okorshenko — CHANGES_REQUESTED** [review]

File: app/code/Magento/Checkout/view/frontend/web/js/checkout-data.js (hunk 1 above)
Comment (arshadpkm) [reply app/code/Magento/Checkout/view/frontend/web/js/checkout-data.js]: Updated in my new commit

**Review by okorshenko — APPROVED** [review]

**Comment by adragus-inviqa** [comment]
If you're going to do it, at least fix the `pullled` typo.

**Comment by ishakhsuvarov** [comment]
Hi @arshadpkm 
Please sync your branch with the latest `develop` and fix merge conflicts
Thanks!

**Comment by arshadpkm** [comment]
@ishakhsuvarov Fixed merge conflicts

**Comment by okorshenko** [comment]
Hi @arshadpkm thank you for your contribution. Your Pull Request has been merged.

**Pull request closed without merging**